### DISCUSSION TITLE
v23.1

### DISCUSSION THREAD

**Discussion by alex-w** [Stellarium/stellarium#3128] [General]
**The installers are located in the "Assets" section at the end of these release notes!**

[![Github Release](https://img.shields.io/github/downloads/Stellarium/stellarium/v23.1/total.svg)](https://github.com/Stellarium/stellarium/releases/v23.1)

## Release notes
The Stellarium Team has released version 23.1.

From now on, releases will have version numbers indicating a year.release scheme. For Windows, the version of the Qt framework used is visible as part of the package name. Details are in the User Guide.

**PLEASE UPDATE!**

This issue mitigates a potential security issue (reported as CVE-2023-28371), where scripts were allowed to write output text and screenshots to other places on users' systems apart from the Stellarium user data directory or configured screenshot directory. Running unknown scripts (which is a Bad Idea(tm) to begin with) could have exploited that by attackers (script authors) writing even command files to vulnerable places. We have not received a report where this would have caused a problem.

However, now users must explicitly allow such behaviour in the settings tab of the Script Console.

Apart from that, this release mostly aimed at fixing issues in the 1.* series, but also brought a few new features:
- Much improved "Solar System observer" and other "Planet observers".
- Solar dates along the ecliptic
- Fixed issues with Observation Lists, and removed need for the extra edit dialog.
- Updated shader code for smoother projection
- Allow texture decimation on the fly for very weak systems like Raspberry Pi 3.
- Solar limb darkening
- Proper shadow edge for satellites

Behind the scenes, many more minor issues were fixed.

**TELESCOPE USERS**

Like in the 1.* series, it seems the TelescopeControl plugin at least on Windows works better in the Qt5-based releases. Just install this if you are affected. We are looking for telescope users with programming skills to help us fixing this issue. Reward possible!

## What's Changed
Full list of changes:
- Added missing include (GH: #2933)
- Added ability to store a local file of objects as bookmark (GH: #2935)
- Added missing file in Babylonian (Seleucid) SC
- Added ability to show the alternative shortcuts in Help dialog (GH: #2944)
- Added B-V color indices for planets and the Sun
- Added B-V indices for few moons
- Added ability to change the font size and color of the landscape labels from the GUI (GH: #2941)
- Added ability to visualize heliocentric ecliptic distances for bright comets in AstroCalc/Positions/HEC tool
- Added an option to use supersampling for shadows on the Moon
- Added implementation of the limb darkening for the Sun (GH: #3017)
- Added Methuselah star
- Added support 32-wind compass rose (GH: #3036)
- Added decorative textures for the observer used in the location panel
- Added accurate ecliptic calendar marks (GH: #3057)
- Added ability to settings fps in the GUI (GH: #3073)
- Added a workaround for poor NPOT mipmap implementations
- Added "Now" button into AstroCalc/Ephemeris tool (GH: #3012)
- Added discovery info for Messier objects (GH: #3084)
- Added phase name and age to Moon's infoMap (GH: #3088)
- Added a check for screenshot file format
- Added proper name for Arp 239
- Added proper name for PGC 1436754 from SIMBAD
- Added support MIT license for sky cultures
- Added GUI to manually entering FOV (GH: #3093, #3013)
- Added ability to toggle ignore date test for orbital data in AstroCalc/Ephemeris tool (GH: #3099)
- Added explicit text message for "satellite below horizon" (GH: #3098)
- Added a confirmation warning to overwrite an existing list on import in Observing lists 
- Added support for masses of the major planets
- Added masses for major planets
- Fixed umbra circles in Satellites plugin (GH: #2510)
- Fixed plotting markers of ephemerides (GH: #2951)
- Fixed the performance penalty: enabled compression of high frequency events (GH: #2778)
- Fixed off-by-one error in planetary rings mesh generator (GH: #2775)
- Fixed compilation on Qt configured for OpenGL ES2 (GH: #2970)
- Fixed bookmarks for minor planets in Solar System Editor plugin (GH: #2958)
- Fixed the wrong transformation of the Moon's color texture values
- Fixed solar eclipse rendering of the Moon in a better way
- Fixed default distance of "observer" to the parent (GH: #2911)
- Fixed handling of lighting factor when rendering lunar eclipse
- Fixed AstroCalc/RTS tool (GH: #3028)
- Fixed another missing timezone
- Fixed location changes via Search panel and "Goto" Action
- Fixed return to default location (with default landscape)
- Fixed other place where location name is printed in bottom bar
- Fixed pixelized panel icons in screenshots (GH: #2899)
- Fixed a possible security issue
- Fixed DSO filters (GH: #3048)
- Fixed screenshot size
- Fixed a potential security issue when storing screenshots
- Fixed scaling of screen geometry when setting window position (GH: #3102)
- Fixed limits for altitudes in Satellites plugin
- Fixed working the filter in shortcut dialog for Qt5-based builds (GH: #2939)
- Fixed storing DSO with designations in Observing lists (GH: #2802)
- Fixed the type/objtype bug in Observing lists 
- Fixed creation of default empty list at start in Observing lists 
- Fixed behaviour for 'Reload the sky culture' feature
- Fixed a crash when observer is on the spaceship (GH: #3115)
- Changed GUI of Satellites plugin: disable a bad implementation of fixed shadow altitude (GH: #3100)
- Changed GUI: better button labels
- Changed core: set line and file numbers in shader prefixes
- Changed core: make ring shadows smoother
- Changed core: improve quality of planetary ring edges
- Changed core: make sure that textures being loaded don't exceed OpenGL size limits (GH: #2991)
- Changed core: avoid blowing out highlights on the lunar surface
- Changed core: use a ratio of FoV and sphere scale when calculating brightness coefficient for the Moon
- Changed core: allow more threads for texture loading
- Changed core: do computations for the planets in linear color, convert to sRGB in the end
- Changed core: move Lambert BRDF from vertex shader to fragment shader
- Changed core: make terminator of planets with atmosphere not so sharp
- Changed core: allow texture decimation (GH: #3009)
- Changed core: simplify checking whether planet being drawn in Planet::draw3dModel is the Sun or the Moon
- Changed core: simplify checking whether current location is Earth in Planet::draw3dModel
- Changed core: draw solar corona before the sphere, regardless of halo settings
- Changed core: make rot_obliquity=0 for all observers
- Changed core: do the computation of lunar eclipse colors in nonlinear color (GH: #3035)
- Changed core: enable mip mapping for old-style-landscape textures)
- Changed core: switch old-style landscape to fragment-shader-based direction computation (GH: #3045)
- Changed core: allow "goto" action to observers
- Changed core: inhibit drawing of a few grids/markers when on Observer pseudo-planet
- Changed core: inhibit location name if observing from Observer pseudo-planet
- Changed core: inhibit landscape and atmosphere drawing when on an "observer" pseudo-planet
- Changed core: make it possible to restrict texture size of GUI panel button pixmaps
- Changed core: softened inner edge of earth penumbra slightly
- Changed core: avoid creation of default user dir if CLI override given (GH: #3079)
- Changed core: deprecated a redundant command line option)
- Changed core: force redraw of dynamic resolution atmosphere on window resize
- Changed core: use config settings screen_* in physical (unscaled) pixels (GH: #3077)
- Changed core: render ShowMySky atmosphere calculating view direction per fragment
- Changed core: tweak screen intersect callback (GH: #3059)
- Changed core: disable culling for ZodiacalLight and MilkyWay
- Changed core: ensure screenshot format initialisation sets supported format
- Changed Satellites plugin: refactoring the code (simplification)
- Changed Satellites plugin: reduced the minimal entered altitude for apogee/perigee filters (GH: #3004)
- Changed Satellites plugin: switched to use WGS-84 data in whole Satellites plugin
- Changed scripting engine: avoid empty filenames for screenshots (GH #3111)
- Changed scripting engine: also make writing to absolute paths a GUI option (GH #3111)
- Changed scripting engine: require user consent to let scripts store screenshots to paths given in the script (GH #3111)
- Changed encoding format in Korean.isl (GH: #2992)
- Updated GUI: losslessly recompress PNGs to save space (GH: #3087)
- Updated orbital elements for minor bodies of Solar system
- Updated default list of locations
- Updated default catalog of satellites
- Updated default catalog of exoplanets
- Updated default catalog of pulsars
- Updated qxlsx dependencies (GH: #2979)
- Updated XML data: use generic-icon in mime file (GH: #3011)
- Updated list of languages
- Updated default bookmarks for Solar System Editor plugin
- Updated translations of sceneries
- Updated translations of landscapes
- Updated translations of sky cultures
- Removed the clamping kludge from the Oren-Nayar function
- Removed NGC 5976A designation (GH: #2967)
- Removed checking architectures in Windows installer (GH: #2962)

## Packages notes
package | note
------------|---------
stellarium-23.1.tar.gz | Stellarium 23.1 for UNIX/Linux (source code). <br />__MD5:__ 7dde5f42e434fff6aa622cd239385709 <br />__SHA256:__ c0d519ef44e78c0609b576c068e1ef2ad09e635d4ff73de7d41bb8c7323bad27
stellarium-23.1.tar.bz2 | Stellarium 23.1 for UNIX/Linux (source code). <br />__MD5:__ 613f226b511b876d0bf45a64a28ccb6e <br />__SHA256:__ 8d8190abd4ee83d94f7c8f2fdb4d41d98090acd0eb401318f73867368c4adedd
stellarium-23.1.tar.xz | Stellarium 23.1 for UNIX/Linux (source code). <br />__MD5:__ ef30b1518ee11e2d6aec364e4533ee22 <br />__SHA256:__ f8e305f41c8e8c21c90b803189e5ef34e5c5f516ab714f35a50a76e536a8fe17
stellarium-23.1-qt5-win32.exe | Stellarium 23.1 for Windows (32-bit; Windows 7+) based on Qt 5.12.6 (Visual Studio 2017 Community) with media support. <br /> __MD5__: 3e6ec7751b0d73b28e016af408d8528c<br />__SHA256__: 2ba220d881b9a64c8e5bbc93997f2ea5912aca42a50b7215d8416557a338991d
stellarium-23.1-qt5-win64.exe | Stellarium 23.1 for Windows (64-bit; Windows 7+) based on Qt 5.12.6 (Visual Studio 2017 Community) with media support. <br /> __MD5__: 0a1728dfc4992c1783d7e328e0debcd2<br />__SHA256__: de2972ba77d9dd1c61e349de3bc112345e7e9797acad0b45f5ad10c202a26afc
stellarium-23.1-qt6-win64.exe | Stellarium 23.1 for Windows (64-bit; Windows 10+) based on Qt 6.4.2 (Visual Studio 2019 Community) with media support. <br /> __MD5__: 9ce10062b84ec0539e36572eee6e6120 <br />__SHA256__: 24f17311aacd464ce6cc600f873f870d73dd5b68c4e5b84573e69b80a57278d9
Stellarium-23.1-qt6-macOS.zip | Stellarium 23.1 for macOS (universal; macOS 11.0+) based on Qt 6.4.3 (AppleClang 14.0.0) with media support. <br /> __MD5__: 921da06ec82c52970f94cb4591eaa293 <br />__SHA256__: 4700f8da7b523b9fdc03222272f26dfbb3d9a94e1bab560868d70e90c601a309
Stellarium-23.1-qt5-arm64.zip | Stellarium 23.1 for macOS (arm64; macOS 10.14+) based on Qt 5.15.8 (AppleClang 14.0.0) with media support. <br /> __MD5__: cdcf21224d65493ea2704a6890fe6672 <br />__SHA256__: f8303085ca0c2e53f32dd1b7c97ca1cddca679ce4d22b87953a5b0a888b70658
Stellarium-23.1-qt5-x86_64.zip | Stellarium 23.1 for macOS (x86_64; macOS 10.14+) based on Qt 5.12.12 (AppleClang 11.0.0) with media support. <br /> __MD5__: 81572a1445c3e65aa84c02c3ada6e6bf <br />__SHA256__: ecd5d3ae86d8c69fa956b0e85fd3465acb8ec4ff5589bc95863dc711b80d534f
Stellarium-23.1-qt5-x86_64.AppImage | Stellarium 23.1 for Linux (amd64; glibc 2.31) based on Qt 5.12.8 (GCC 9.4.0) with media support. <br /> __MD5__: 8cbd9032c7c10b9e5cc49e517273bcc1 <br />__SHA256__: 8f59f09c8b156f9384d814995d039b734aa9613beedee845dc74849194c23222

<hr /><em>This discussion was created from the release <a href='https://github.com/Stellarium/stellarium/releases/tag/v23.1'>v23.1</a>.</em>

**Comment by ilkant** [comment]
Please, update also stellarium.org texts from Transifex. I have done some fixes there.

**Reply by alex-w** [reply]
> Please, update also stellarium.org texts from Transifex. I have done some fixes there.

It was updated yesterday (50%+ of completion)

**Comment by ilkant** [comment]
When I download Stellarium-23.1-qt5-x86_64.AppImage and start it in Kubuntu 22.10, it works but online query diallog opeing chrashes whole program. If I pull source code and compile Stellarium on my computer, the online query dialog/winodws works ok. There are something wrong with AppImage. That has occureed with previous versions of Stellarium too. But no problem. I use on my computer compiled Stellarium.

And thanks! There have fixed lots of bugs and created new features.

**Reply by alex-w** [reply]
> When I download Stellarium-23.1-qt5-x86_64.AppImage and start it in Kubuntu 22.10, it works but online query diallog opeing chrashes whole program. If I pull source code and compile Stellarium on my computer, the online query dialog/winodws works ok. There are something wrong with AppImage. That has occureed with previous versions of Stellarium too. But no problem. I use on my computer compiled Stellarium.

Please fill the bug report and attach log.txt from Stellarium

**Comment by AstroJimF** [comment]
I have been using Stellarium for years on mobile devices, but installed this morning on my laptop as I am trying to do something different. 

Anyway, I have had the same problem no matter which of the latest Windows versions I use. All 23.1 but both 64bit and 32Bit (X86).

There is no Star names present, and S or Alt S does nothing to produce them, thus I have blank constellations - I do not even get the Greek letters for the stars let alone proper names - which is what I need to do what my intent is - Education

The laptop I am using for this is a few years old now - if it ain't broke - don't "fix" it is my attitude. Below are the details of the machine and the graphics information - including driver details. I do not know if this is an issue with my machine now getting old and I should use an older version of Stellarium, or if there is something in the Stellarium software causing the issue - my tablet is running a version about 2 years old and that runs fine. 

Device name	Lenovo_Yoga_Touch
Processor	Intel(R) Core(TM) i5-6200U CPU @ 2.30GHz   2.40 GHz
Installed RAM	16.0 GB (15.9 GB usable)
Device ID	9E32B9D5-0D75-496F-8EC3-F0A6EB4B705E
Product ID	00330-80000-00000-AA400
System type	64-bit operating system, x64-based processor
Pen and touch	Touch support with 10 touch points 
Edition	Windows 10 Pro
Version	21H2
Installed on	‎02/‎04/‎2021
OS build	19044.2728
Experience	Windows Feature Experience Pack 120.2212.4190.0
Firmware Details: 10.0.19.41.1 (Dated 21/06/2006) !!! (Wow – MS are behind that curveball here!)

Graphics Details.
Intel HD Graphics 520
Driver version: 23.20.16.4973 (Latest Available)
Driver date: 28/02/2018

AMD Radeon R5 M330 - Primary
Driver version: 27.20.1034.6 (latest version)
Driver date: 21/08/2020

**Reply by 10110111** [reply]
Note that mobile and desktop versions of Stellarium are two completely separate pieces of software, with zero shared code (only some textures are common). So the differences in behavior are to be expected.

Still, the bright stars like Sirius and Arcturus should have labels when they are turned on. Other stars you can click to get some info about them.

**Reply by AstroJimF** [reply]
Thank you for the response, but not actually helpful.

**Reply by gzotti** [reply]
Please see section 4.4.1 of the User Guide and its Figure 4.10. (And maybe read or at least skim chapters 2-4 if the desktop version is new to you.)

**Reply by AstroJimF** [reply]
Not any help at all. I have used Stellarium on a PC before for planning purposes so this is not new to me, but 23.1 does not appear to work properly with the latest Windows. I installed it on an old machine running an original W10 Home copy - works fine - installed on my Workstation running Windows 10 Pro, but lacking the latest updates - it's not connected to the Internet - and it works fine - but on this machine, running W10 Pro with all the updates, it does not want to work - my money is on the code.

**Reply by 10110111** [reply]
"Not any help at all", "not actually helpful" — these replies are also not helpful in identifying your actual issue: what you expected, what actually happens, how it behaved before if it worked better and in which version, what exactly looks different etc.

**Reply by AstroJimF** [reply]
I must assume you did not read my initial post which explains the issue clearly and concisely - never mind, I will find the time to figure it out myself

**Reply by gzotti** [reply]
I confirm 23.1 works on the latest Windows 11. S switches the stars, Alt-S switches star labels, and the section cited above provides information and a figure which shows a slider to increase number of displayed star names. Your system should work as well. "Should" is the key. If you suspect a bug, let me quote from dozens of earlier reports: Please fill the bug report and attach log.txt from Stellarium, reading of which is all of remote diagnostics we have.
Last idea before reading your log.txt: check which sky culture you have activated.

**Reply by 10110111** [reply]
I think your problem might be solvable by the slider in *View*→*Sky*→*Labels and Markers* control. When I move this slider to the rightmost position, I get loads of labels: for each star displayed.

**Reply by alex-w** [reply]
@AstroJimF please fill a bug report for desktop edition of Stellarium and attach `log.txt` and `config.ini` files

**Comment by carlsd58** [comment]
Upon installing the new 23.1 version of Stellarium I've noticed the 64 bit installer has a default install location of C:\Program Files (x86)\Stellarium which is normally where 32 bit programs are installed. Up until now, the 64 bit version has always installed in C:\Program Files\Stellarium.

Is this by design or is there some error in the installer? I can manually change the install location of course and when I do change it to C:\Program Files\Stellarium I'm told Stellarium is already installed there and do I wish to install there anyway. 

I click OK and the installation proceeds normally and Stellarium appears to work fine after I'm finished. Is it OK to install over the top of the old version with this new release? Been doing it for a few years now and I've never had a problem doing that. I'm using the latest release of Windows 10 with all of the updates installed.

Thank you for the excellent program, by the way.

**Reply by alex-w** [reply]
Please see bug #3130

**Reply by carlsd58** [reply]
Noted. Thank you Alexander.

**Reply by cwilly8** [reply]
I'm a bit slow...  What is the answer to carlsd58 concern?  Just let the installer put the Windows 64 version of 23.1 in the x86 folder?  I read the comments in bug #3130 but I'm not sure what the final answer is.  Thanks.

**Reply by gzotti** [reply]
It's just cosmetics. If you are worried, edit the target directory presented during installation and erase "(x86)".

**Reply by cwilly8** [reply]
Thanks for the quick reply, gzotti!

**Reply by Aron-Love** [reply]
> It's just cosmetics. If you are worried, edit the target directory presented during installation and erase "(x86)".

Can this be done silently from the command line?

**Reply by gzotti** [reply]
I don't know. Do you run a Windows installer from the command line? 
This is a cosmetic issue that will go away in 23.2. I'd turn to more interesting stuff.

**Reply by Aron-Love** [reply]
> I don't know. Do you run a Windows installer from the command line? This is a cosmetic issue that will go away in 23.2. I'd turn to more interesting stuff.

OK. Our environment is mixed with Windows 10/11. I use SCCM for application management. For this release, I added a line to my upgrade script which looks for "C:\Program Files (x86)\Stellarium\unins000.exe" and "C:\Program Files\Stellarium\unins000.exe" and then runs the correct uninstaller silently. So when future updates change the default install folder, I'll be ready. Thanks.

**Comment by dg050415** [comment]
vs2019 community +qt5.15.2+stellarium-23.1
I compiled the source code, successfully run, but press R can not see the constellation picture.
[log.txt](https://github.com/Stellarium/stellarium/files/11595613/log.txt)
